command/command.go

```diff
@@ -54,46 +56,161 @@ func IsForce() bool {
+				if err != nil {
+					return err
+				}
+				ctxStr = items[i].Name
+			}
+		}
 	}
 
-	flags := make(map[string]cli.Flag)
-	for _, f := range ctx.Command.Flags {
-		name := strings.Split(f.GetName(), ",")[0]
-		flags[name] = f
+	if ctxStr != "" {
+		if err := step.SwitchCurrentContext(ctxStr); err != nil {
+			return err
+		}
 	}
 
-	for _, name := range ctx.FlagNames() {
-		if ctx.IsSet(name) {
-			continue
+	var m map[string]interface{}
+	if step.GetCurrentContext() == nil {
+		configFile := ctx.GlobalString("config")
+		if configFile == "" {
+			configFile = filepath.Join(step.BasePath(), "config", "defaults.json")
 		}
 
-		// Skip if EnvVar == IgnoreEnvVar
-		if f, ok := flags[name]; ok {
-			if getFlagEnvVar(f) == IgnoreEnvVar {
-				continue
+		_, err := os.Stat(configFile)
+		switch {
+		case os.IsNotExist(err):
+			return nil
+		case err != nil:
+			return err
+		default:
+			b, err := ioutil.ReadFile(configFile)
+			if err != nil {
+				return nil
+			}
+			m = make(map[string]interface{})
+			if err := json.Unmarshal(b, &m); err != nil {
+				return errors.Wrapf(err, "error parsing %s", configFile)
 			}
 		}
+	} else {
+		if strings.HasPrefix(fullCommandName, "ca bootstrap-helper") {
+			return nil
+		}
+
+		authorityMap := make(map[string]interface{})
+		authorityConfigFile := filepath.Join(step.Path(), "config", "defaults.json")
+		_, err := os.Stat(authorityConfigFile)
+		switch {
+		case os.IsNotExist(err):
+			break
+		case err != nil:
+			return err
+		default:
+			b, err := ioutil.ReadFile(filepath.Join(authorityConfigFile))
+			if err != nil {
+				return errs.FileError(err, authorityConfigFile)
+			}
 
-		if v, ok := m[name]; ok {
-			ctx.Set(name, fmt.Sprintf("%v", v))
+			if err := json.Unmarshal(b, &authorityMap); err != nil {
+				return errors.Wrapf(err, "error parsing %s", authorityConfigFile)
+			}
+		}
+
+		profileMap := make(map[string]interface{})
+		profileConfigFile := filepath.Join(step.ProfilePath(), "config", "defaults.json")
+		_, err = os.Stat(profileConfigFile)
+		switch {
+		case os.IsNotExist(err):
+			break
+		case err != nil:
+			return err
+		default:
+			b, err := ioutil.ReadFile(profileConfigFile)
+			if err != nil {
+				return nil
+			}
+			if err := json.Unmarshal(b, &profileMap); err != nil {
+				return errors.Wrapf(err, "error parsing %s", profileConfigFile)
+			}
+		}
+
+		// Combine authority and profile maps such that profile values take precedence.
+		for k, v := range authorityMap {
+			if _, ok := profileMap[k]; !ok {
+				profileMap[k] = v
+			}
+		}
+		m = profileMap
+	}
+
+	for _, f := range ctx.Command.Flags {
+		// Skip if EnvVar == IgnoreEnvVar
+		if getFlagEnvVar(f) == IgnoreEnvVar {
+			continue
+		}
+
+		for _, name := range strings.Split(f.GetName(), ",") {
+			name = strings.TrimSpace(name)
+			if ctx.IsSet(name) {
+				break
+			}
+			// Set the flag for the first key that matches.
+			if v, ok := m[name]; ok {
+				ctx.Set(name, fmt.Sprintf("%v", v))
+				break
+			}
```

Comment:
All this logic was different before, starting that the initial range was over `ctx.FlagNames()` that skips `help` and that we were only supporting the main name for the flag and not the aliases.